### PR TITLE
Make better use of loom

### PR DESCRIPTION
**Pull request by jonhoo**
This actually uses loom types in all `#[cfg(test)]` cases (which we use in `tests/fuzz_*` by directly including the impls as modules).

### CODE REVIEW

**Comment by jonhoo** [comment]
The fuzz tests currently fail due to broken scheduler behavior in `loom` likely due to the use of `generator-rs`

File: tokio-sync/src/watch.rs [hunk 1]
```diff
@@ -53,16 +53,20 @@
...
 use core::task::{Context, Poll};
 use fnv::FnvHashMap;
 use futures_util::future::poll_fn;
 use std::ops;
-use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering::SeqCst;
-use std::sync::{Arc, Mutex, RwLock, RwLockReadGuard, Weak};
+
+// We'd like to use loom for these too, but loom doesn't have them yet
+use std::sync::{Arc, RwLock, RwLockReadGuard, Weak};
```
Comment (pickfire) [inline tokio-sync/src/watch.rs]: I believe loom already have these?

https://github.com/carllerche/loom/blob/986ca4234400b3232a495943d99244d6af8db37d/src/sync/arc.rs#L5-L9

**Comment by carllerche** [comment]
Closing due to inactivity, though we should probably investigate more...

**Comment by carllerche** [comment]
Hey @jonhoo, I believe I fixed the bug in loom that was causing this. Anyway to reboot this Pr?

**Comment by jonhoo** [comment]
@carllerche Which parts of it?

**Pull request closed without merging**